test/e2e/functional/e2e_test.go

```diff
@@ -36,17 +35,25 @@ func init() {
 func TestE2E(t *testing.T) {
 	RegisterFailHandler(Fail)
 
-	rr := []Reporter{}
+	_, reporterConfig := GinkgoConfiguration()
+
 	if *junitPath != "" {
 		junitFile := path.Join(*junitPath, "e2e_junit.xml")
-		rr = append(rr, reporters.NewJUnitReporter(junitFile))
+		reporterConfig.JUnitReport = junitFile
 	}
 
-	clients := testclient.New("")
+	RunSpecs(t, "Metallb Operator E2E Suite", reporterConfig)
+}
 
-	if *reportPath != "" {
-		rr = append(rr, k8sreporter.New(clients, OperatorNameSpace, *reportPath))
+var _ = ReportAfterEach(func(specReport types.SpecReport) {
+	if specReport.Failed() == false {
+		return
 	}
 
-	RunSpecsWithDefaultAndCustomReporters(t, "Metallb Operator E2E Suite", rr)
-}
+	if *reportPath != "" {
```

Comment:
how about initializing the reporter object at setup time? Here we re-create it every time a test fail.